test/subarray.jl

```diff
@@ -588,6 +588,7 @@ end
 primitive type UInt48 48 end
 UInt48(x::UInt64) = Core.Intrinsics.trunc_int(UInt48, x)
 UInt48(x::UInt32) = Core.Intrinsics.zext_int(UInt48, x)
+Base.read(io::IO, ::Type{UInt48}) = only(reinterpret(UInt48, read(io, sizeof(UInt48))))
```

Comment:
This must be what's causing the failure --- it reads too few bytes, failing to skip the alignment padding.